File: integrations/typescript/datadog/src/DatadogSink.ts
import type { Sink, WideEventPayload } from '@axora/logger-core';

export class DatadogSink implements Sink {
  async export(event: WideEventPayload): Promise<void> {
    const logEntry: Record<string, unknown> = {
      date: event.timestamp_start,
      status: event.level,
      service: event.service,
      message: event.operation,
      duration: event.duration_ms,
      ...event.context,
      'dd.axora_event_id': event.event_id,
      'dd.env': event.environment,
    };

    if (event.error.message) {
      logEntry.error = {
        kind: event.error.type,
        message: event.error.message,
        stack: event.error.stack,
      };
    }

    process.stdout.write(`${JSON.stringify(logEntry)}\n`);
  }
}

File: integrations/csharp/datadog/DatadogSink.cs
using System.Text.Json;
using Openakta.Logger.Sinks;

namespace Openakta.Logger.Datadog;

public sealed class DatadogSink : ISink
{
    public Task ExportAsync(WideEventPayload @event)
    {
        var logEntry = new Dictionary<string, object?>
        {
            ["date"] = @event.TimestampStart,
            ["status"] = @event.Level,
            ["service"] = @event.Service,
            ["message"] = @event.Operation,
            ["duration"] = @event.DurationMs,
            ["dd.openakta_event_id"] = @event.EventId,
            ["dd.env"] = @event.Environment,
        };

        foreach (var (key, value) in @event.Context)
        {
            logEntry[key] = value;
        }

        if (@event.Error.Message is not null)
        {
            logEntry["error"] = new Dictionary<string, object?>
            {
                ["kind"] = @event.Error.Type,
                ["message"] = @event.Error.Message,
                ["stack"] = @event.Error.Stack,
            };
        }

        Console.Out.WriteLine(JsonSerializer.Serialize(logEntry));
        return Task.CompletedTask;
    }
}
